Improve Ansible content modules - Premigration (#5001)
theforeman/foreman-documentation: guides/common/modules/proc_synchronizing-ansible-collections.adoc

Reviewer: There are some issues related to whitespace in this file. You can check this locally:

````
$ git diff -w master guides/common/modules/proc_synchronizing-ansible-collections.adoc
$ git diff master guides/common/modules/proc_synchronizing-ansible-collections.adoc
````

Author: > There are some issues related to whitespace in this file. You can check this locally:
> 
> ```
> $ git diff -w master guides/common/modules/proc_synchronizing-ansible-collections.adoc
> $ git diff master guides/common/modules/proc_synchronizing-ansible-collections.adoc
> ```

Checked and resolved for this commit. 

I am a bit curious to know how (and when) did these characters entered and what configuration is required not to get this error again. 
For this time, i have used `dos2unix` to fix the error.

:_mod-docs-content-type: PROCEDURE

> [id="synchronizing-ansible-collections_{context}"]

Reviewer: ```suggestion
[id="synchronizing-ansible-collections"]
```

IMO the anchor was OK as is before.

= Synchronizing Ansible Collections

[role="_abstract"]
ifdef::satellite[]
On {Project}, you can synchronize your Ansible Collections from Private Automation Hub, `console.redhat.com`, and other {Project} instances to centrally manage and distribute automation content across your infrastructure.
endif::[]
ifndef::satellite[]
On {Project}, you can synchronize your Ansible Collections from any Ansible Galaxy and other {Project} instances to centrally manage and distribute automation content across your infrastructure.
endif::[]
Ansible Collections will appear on {Project} as a new repository type in the {ProjectWebUI} menu under *Content* after the sync.

> 

Reviewer: ```suggestion
```

+
The URL can be any Ansible Galaxy endpoint.
ifdef::satellite[]
For example, `\https://console.redhat.com/api/automation-hub/`.
endif::[]
ifndef::satellite[]
For example, `\https://galaxy.ansible.com/`.
endif::[]
. Optional: In the *Requirements.yml* field, you can specify the list of collections you want to sync from the endpoint and their versions.
+
If you do not specify a list of collections, the system synchronizes all Ansible collections from the endpoint.
+
[source, yaml]
----
---
collections:
- name: my_namespace.my_collection
  version: 1.2.3
----
+
For more information, see https://docs.ansible.com/ansible/latest/galaxy/user_guide.html#installing-roles-and-collections-from-the-same-requirements-yml-file[Installing roles and collections from the same requirements.yml file] in the _Galaxy User Guide_.
. Optional: Deselect *Sync Dependencies* if you do not want {Project} to resolve and synchronize dependencies.
By default, {Project} synchronizes all required dependencies.
. Authenticate:
* To sync Ansible collections from *Private Automation Hub*:
** Navigate to https://console.redhat.com/ansible/automation-hub/token[Connect to Hub].
Click *Load token*.

> ** Enter the token in the *Auth Token* field in the {ProjectWebUI}.

Reviewer 2: ```suggestion
.. Navigate to https://console.redhat.com/ansible/automation-hub/token[Connect to Hub].
Click *Load token*.
.. Enter the token in the *Auth Token* field in the {ProjectWebUI}.
```

* To sync Ansible collections from `console.redhat.com`:
** Navigate to https://console.redhat.com/ansible/automation-hub/token[Connect to Hub].
Click *Load token*.
** Enter the authentication token in the *Auth Token* field in the {ProjectWebUI}.
** Copy the SSO URL from the SSO URL section.

> ** Enter the SSO URL in the *Auth URL* field in the {ProjectWebUI}.

Reviewer 2: ```suggestion
.. Navigate to https://console.redhat.com/ansible/automation-hub/token[Connect to Hub].
Click *Load token*.
.. Enter the authentication token in the *Auth Token* field in the {ProjectWebUI}.
.. Copy the SSO URL from the SSO URL section.
.. Enter the SSO URL in the *Auth URL* field in the {ProjectWebUI}.
```

* To sync Ansible collections from {Project}, leave both authentication fields blank.
. Click *Save*.
. Navigate to the Ansible Collections repository.
. From the *Select Action* menu, select *Sync Now*.
ifdef::satellite[]

> 

Reviewer: ```suggestion

ifdef::satellite[]
```

[role="_abstract"]
ifdef::satellite[]
On {Project}, you can synchronize your Ansible Collections from Private Automation Hub, `console.redhat.com`, and other {Project} instances to centrally manage and distribute automation content across your infrastructure.
endif::[]
ifndef::satellite[]
On {Project}, you can synchronize your Ansible Collections from any Ansible Galaxy and other {Project} instances to centrally manage and distribute automation content across your infrastructure.
endif::[]
Ansible Collections will appear on {Project} as a new repository type in the {ProjectWebUI} menu under *Content* after the sync.

> .Prerequisites

Author: ```suggestion
```

[role="_abstract"]
ifdef::satellite[]
On {Project}, you can synchronize your Ansible Collections from Private Automation Hub, `console.redhat.com`, and other {Project} instances to centrally manage and distribute automation content across your infrastructure.
endif::[]
ifndef::satellite[]

> On {Project}, you can synchronize your Ansible Collections from any Ansible Galaxy and other {Project} instances.

Reviewer: Please apply the same change from line 8 to line 11 too.

Author: See the next build for changes.

endif::[]
Ansible Collections will appear on {Project} as a new repository type in the {ProjectWebUI} menu under *Content* after the sync.

.Prerequisites
* To sync Ansible collections from Red Hat Private Automation Hub, you have obtained an authentication token from https://console.redhat.com/ansible/automation-hub/token[Connect to Hub].

> * To sync Ansible collections from `console.redhat.com`, you have obtained an authentication token and the SSO URL from https://console.redhat.com/ansible/automation-hub/token[Connect to Hub].

Reviewer 2: Is this change strictly necessary for CQA? These instructions are the result of a bug report: https://redhat.atlassian.net/browse/SAT-40367 is the Jira, https://github.com/theforeman/foreman-documentation/pull/4660 is the PR. I'm concerned that implementing this change will effectively reverse that bug fix.

Author: > Is this change strictly necessary for CQA? These instructions are the result of a bug report: https://redhat.atlassian.net/browse/SAT-40367 is the Jira, #4660 is the PR. I'm concerned that implementing this change will effectively reverse that bug fix.

As far as CQA rules, it is necessary to remove all instructional content from the prerequisites. [See row number 63 under Pre-migration](https://docs.google.com/spreadsheets/d/1xeNvhdXPi5IClgFzPbIOH9IKp2Y6i9devr6I--2Rd3g/edit?gid=503904153#gid=503904153). The quality level for this is decided to **Required/non-negotiable**.

Reviewer 2: Yes, but not without a replacement. Prerequisites shouldn't include steps but if a prerequisite includes a step, it should be moved somewhere else so we don't lose valuable information. Can you please find a way to keep this information but perhaps include it in the procedure itself? Hopefully the Jira report and the PR link will also explain what the original problem was that the existing prerequisites have solved for users.

Author: @aneta-petrova do you know any existing procedure which has this information documented? We can redirect users there by adding a cross reference (xref or link).  I was going to suggest adding it in the first few steps but I think this is not possible too. We already have another rule under [Quality](https://docs.google.com/spreadsheets/d/1xeNvhdXPi5IClgFzPbIOH9IKp2Y6i9devr6I--2Rd3g/edit?gid=503904153#gid=503904153) tab to keep number of steps under 10. See row number 53 under Quality.

Reviewer 2: Unfortunately, no, I'm not aware of any procedure we could link to.

Author: > Unfortunately, no, I'm not aware of any procedure we could link to.

Ok. I have suggested something different. See if this works.

Reviewer 2: I'm not too sure that `Connect to Hub > Load Token` addresses either the CQA requirement or the bug fix.

* CQA says prerequisites can't contain steps: The latest version of the prerequisites are still asking users to take steps, except that while before those steps were clearly documented, now they're hidden within a link label.

* The bug fix was about providing clear instructions on how to obtain tokens: Merging the instructions with the link labels now effectively means those instructions are unclear again.

Other solutions that come to my mind are:

* Move the instructions to the procedure and accept that it already has more than 10 steps so this wouldn't break anything that's not already broken. Then you could raise a bug for the "quality" issues and report it there for someone to fix the whole procedure later.
* Move the instructions to the procedure and make sure it doesn't have more than 10 steps. The CQA requirement talks about "steps" but not "substeps". So what if there's a way to reorganize the procedure so that it has 10 steps maximum, and those steps present substeps grouped in a logical and user-friendly way?
* (EDIT:) Move the instructions to the procedure and then split the procedure into multiple smaller procedures.

Author: > I'm not too sure that `Connect to Hub > Load Token` addresses either the CQA requirement or the bug fix.
> 
> * CQA says prerequisites can't contain steps: The latest version of the prerequisites are still asking users to take steps, except that while before those steps were clearly documented, now they're hidden within a link label.
> * The bug fix was about providing clear instructions on how to obtain tokens: Merging the instructions with the link labels now effectively means those instructions are unclear again.
> 
> Other solutions that come to my mind are:
> 
> * Move the instructions to the procedure and accept that it already has more than 10 steps so this wouldn't break anything that's not already broken. Then you could raise a bug for the "quality" issues and report it there for someone to fix the whole procedure later.

Ok. I think this option sounds better to me. I will fix it in another commit and then resolve this comment. 

> * Move the instructions to the procedure and make sure it doesn't have more than 10 steps. The CQA requirement talks about "steps" but not "substeps". So what if there's a way to reorganize the procedure so that it has 10 steps maximum, and those steps present substeps grouped in a logical and user-friendly way?
> * (EDIT:) Move the instructions to the procedure and then split the procedure into multiple smaller procedures.

Author: See the next build for changes.

> * To sync Ansible collections from `console.redhat.com`, you have obtained an authentication token and the SSO URL from https://console.redhat.com/ansible/automation-hub/token[Connect to Hub].

Reviewer: I am unsure if it would be better to combine this:

```suggestion
* If you want to synchronize Ansible collections from Red Hat Private Automation Hub or `console.redhat.com`, you have obtained an authentication token from https://console.redhat.com/ansible/automation-hub/token[Connect to Hub].
```

One downside is that the users that want to use console.redhat.com have to continue reading after "If you want to synchronize Ansible collections from Red Hat Private Automation Hub". What do you think?

Author: > I am unsure if it would be better to combine this:
> 
> One downside is that the users that want to use console.redhat.com have to continue reading after "If you want to synchronize Ansible collections from Red Hat Private Automation Hub". What do you think?

If they are syncing from `console.redhat.com`, they need 2 things; authentication token and SSO URL. So i think it wont work.

> * To sync Ansible collections from `console.redhat.com`, you have an *authentication token* and the *SSO URL*.

Reviewer 2: ```suggestion
```
If users are instructed to obtain the tokens during the procedure, they can't have it ready as a prerequisite before starting the procedure.

. Click the *Repositories* tab, and then click *New Repository*.
. In the *Name* field, enter a name for the repository.
+

> The system automatically populates the *Label* field based on the name.

Reviewer 2: Would you be okay reverting this to passive voice?  Line 39 as well. The IBM Style Guide says that passive voice is perfectly fine when "the system" is performing the action. (per the section "Voice" in the style guide)

Saying "the system automatically populates..." adds unnecessary words or even makes users wonder what is that unspecified system ("is it an outside application triggering the change?" 👀). The passive version keeps it a bit cleaner and focused on the Label field itself. That's why I'd prefer to keep the original for optimal user-friendliness.

> The system automatically populates the *Label* field based on the name.

Author: ```suggestion
The *Label* field is populated automatically based on the name.
```

ifndef::satellite[]
For example, `\https://galaxy.ansible.com/`.
endif::[]
. Optional: In the *Requirements.yml* field, you can specify the list of collections you want to sync from the endpoint and their versions.
+

> If you do not specify a list of collections, the system synchronizes all Ansible collections from the endpoint.

Reviewer: ```suggestion
If you do not specify a list of collections, {Project} synchronizes all Ansible collections from the endpoint.
```

ifndef::satellite[]
For example, `\https://galaxy.ansible.com/`.
endif::[]
. Optional: In the *Requirements.yml* field, you can specify the list of collections you want to sync from the endpoint and their versions.
+

> If you do not specify a list of collections, the system syncs all collections from the endpoint.

Reviewer: ```suggestion
If you do not specify a list of collections, the system synchronizes all Ansible collections from the endpoint.
```

. Optional: Deselect *Sync Dependencies* if you do not want {Project} to resolve and synchronize dependencies.
By default, {Project} synchronizes all required dependencies.
. Authenticate:
* To sync Ansible collections from *Private Automation Hub*:

> ** Navigate to https://console.redhat.com/ansible/automation-hub/token[Connect to Hub] > click *Load token*.

Reviewer 2: ```suggestion
** Navigate to https://console.redhat.com/ansible/automation-hub/token[Connect to Hub].
Click *Load token*.
```

> ** Navigate to https://console.redhat.com/ansible/automation-hub/token[Connect to Hub] > click *Load token*.

Reviewer 3: These list items are sub-steps, so I'd go with a numbered list.
```suggestion
.. Navigate to https://console.redhat.com/ansible/automation-hub/token[Connect to Hub].
.. Click *Load token*.
```

> ** Enter the token in the *Auth Token* field.

Reviewer 2: Where? :) Right now, users are at the Connect to Hub page (because that's where the previous substep has directed them) but I don't think that's where they are supposed to enter the token.

Author: I do not have access to the product. If you have access, can you add the missing details, pls?

Reviewer 3: You don't need access to the product, you can just say that it's in the Foreman/Satellite web UI.
```suggestion
** Enter the token in the *Auth Token* field in the {ProjectWebUI}.
```

* To sync Ansible collections from `console.redhat.com`:

> ** Navigate to https://console.redhat.com/ansible/automation-hub/token[Connect to Hub] > click *Load token*.

Reviewer 2: ```suggestion
** Navigate to https://console.redhat.com/ansible/automation-hub/token[Connect to Hub].
Click *Load token*.
```

> ** Enter the authentication token in the *Auth Token* field.

Reviewer 2: Same as above -- where?

Author: > Same as above -- where?

I do not have access to the product. If you have access, can you add the missing details, pls?

Reviewer 3: ```suggestion
** Enter the authentication token in the *Auth Token* field in the {ProjectWebUI}.
```

** Copy the SSO URL from the SSO URL section.

> ** Enter the SSO URL in the *Auth URL* field.

Reviewer 2: The SSO URL section is at https://console.redhat.com/ansible/automation-hub/token, but the Auth URL field is in the Foreman/Satellite web UI. So basically same as above -- this should somehow specify where users perform the actions.

Author: > The SSO URL section is at https://console.redhat.com/ansible/automation-hub/token, but the Auth URL field is in the Foreman/Satellite web UI. So basically same as above -- this should somehow specify where users perform the actions.

I do not have access to the product. If you have access, can you add the missing details, pls?

Reviewer 3: ```suggestion
** Copy the SSO URL from the SSO URL section.
** Enter the SSO URL in the *Auth URL* field in the {ProjectWebUI}.
```
